File: app/src/main/java/com/example/smarthome/ligths/models/Light.java
package com.example.smarthome.ligths.models;

import android.os.strictmode.CredentialProtectedWhileLockedViolation;

import com.example.smarthome.WifiDevice;
import com.google.gson.annotations.SerializedName;

public class Light extends WifiDevice {

    private Long id;
    private short red;
    private short green;
    private short blue;
    private short intensity;

    public Light(){}

    public Light(String name, String ip, Boolean on, short red, short green, short blue, short intensity){
        super(name, ip, on);
        this.red = red;
        this.green = green;
        this.blue = blue;
        this.intensity = intensity;
    }

    public Long getId() {
        return id;
    }

    public short getRed() {
        return red;
    }

    public short getGreen() {
        return green;
    }

    public short getBlue() {
        return blue;
    }

    public short getIntensity() {
        return intensity;
    }
}

File: app/src/main/java/com/example/smarthome/ligths/models/Light.kt
package com.example.smarthome.ligths.models

import com.example.smarthome.R
import com.example.smarthome.WifiDevice
import com.example.smarthome.adapter.AdapterConstants
import com.example.smarthome.adapter.ViewType

data class Light(
        val id: Long?,
        val red: Short,
        val green: Short,
        val blue: Short,
        val intensity: Short
) : WifiDevice(), ViewType {
    constructor(name: String, ip: String, on: Boolean,
                red: Short, green: Short, blue: Short, intensity: Short)
        :this(null, red, green, blue, intensity){
        super.setName(if (name != "") name else "No name")
        super.setIp(ip)
        super.setOn(on)
    }

    override fun getViewType(): Int {
        return 0 //TODO R.layout.item_light
    }
}